app/services/data_set_documentation.rb

```diff
@@ -1,30 +1,34 @@
 class DataSetDocumentation
   def self.for(klass)
-    name = klass.name.demodulize.underscore
-
-    begin
-      spec = YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
-    rescue Errno::ENOENT
-      return false
-    end
+    export_documentation = load_export_documentation(klass)
+    common_columns = load_common_columns
 
-    common_columns = YAML.load_file(Rails.root.join('app/exports/_common_columns.yml'))
-    used_common_columns = common_columns.slice(*spec['common_columns'])
-    used_common_columns.merge(spec['custom_columns'] || {})
+    used_common_columns = common_columns.slice(*export_documentation['common_columns'])
+    used_common_columns.merge(export_documentation['custom_columns'] || {})
   end
 
   def self.shadowed_common_columns(klass)
+    export_documentation = load_export_documentation(klass)
+
+    common_columns = load_common_columns.keys.map(&:to_sym)
+    custom_columns = export_documentation['custom_columns'].keys.map(&:to_sym)
+
+    custom_columns & common_columns
+  end
+
+  def self.load_export_documentation(klass)
     name = klass.name.demodulize.underscore
 
     begin
-      spec = YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
+      return YAML.load_file(Rails.root.join("app/exports/#{name}.yml"))
     rescue Errno::ENOENT
       return false
```

Comment:
I missed that this `false` was being relied upon to exit early from this method and this refactor broke that mechanism. I've reverted those changes now and I plan to leave it as it is. 

Useless there are any other suggestions?

Reply:
See other comment for context

Reply:
If we expect everything to have documentation, should this throw an exception if the file doesn't exist?

Reply:
Yes, I think that is a nice solution, once everything has documentation. I will add that in as part of a later PR.